src/main/java/randoop/main/GenTests.java

```diff
@@ -516,6 +522,75 @@ public boolean handle(String[] args) {
+
+        // How many tests an operation occurs in.
+        // Tally occurrences of operations in all tests.
+        // Each method is counted exactly once if it appears in a sequence test.
+        Map<TypedOperation, Integer> testOccurrences =
+            tallyOperationsInSequences(regressionSequences);
+
+        // How many flaky tests an operation occurs in.
+        Map<TypedOperation, Integer> flakyOccurrences;
+
+        // TODO: cxing handle Error Test Sequence tallying.
+        // tallyOperationsInSequences(testOccurrences, explorer.getErrorTestSequences());
+
+        List<ExecutableSequence> flakySequences = new ArrayList<>();
+        for (String flakyTestName : flakyTestNames) {
+          int testNum = Integer.parseInt(flakyTestName.substring(4)); // length of "test"
+          // Tests start at 001
+          ExecutableSequence flakySequence = regressionSequences.get(testNum - 1);
+          flakySequences.add(flakySequence);
+        }
+
+        // Tally occurrences of operations in flaky tests.
+        // Each method is counted exactly once if it appears in a flaky test.
+        flakyOccurrences = tallyOperationsInSequences(flakySequences);
+
+        // tf-idf metric
+        // Our heuristic for probability of whether a method is flaky
+        //  is equal to the ratio of the number of flaky tests a method M occurs
+        //  in divided by the number of total tests M occurs in.
+        // The result is multiplied by 100.0 to obtain a percentage.
+
+        // Maps from a fully-qualified method signature to the above
+        //  percentage.
+        Map<String, Double> methodFlakyPercentage = new HashMap<>();
+
+        for (TypedOperation to : testOccurrences.keySet()) {
+          double flakyMethodOccurrences =
+              flakyOccurrences.containsKey(to) ? flakyOccurrences.get(to) : 0;
+          methodFlakyPercentage.put(
+              to.toParsableString(), flakyMethodOccurrences / testOccurrences.get(to));
+        }
+
+        // Sort in descending order by value.
+        List<Entry<String, Double>> sortedMethodsByFlakiness =
+            new ArrayList<>(methodFlakyPercentage.entrySet());
+        sortedMethodsByFlakiness.sort(reverseOrder(Entry.comparingByValue()));
+
+        System.out.println();
+        System.out.println("Flaky tests were generated. The following methods,");
+        System.out.println("in decreasing order of likelihood, are the most likely culprits.");
+        System.out.println("Please determine whether the following methods may exhibit");
+        System.out.println("different behavior on separate runs of the same code.");
+        // TODO cxing: add nmrd-blacklist comment suggestion for user.
+        System.out.println();
+
+        // Output methods by ranking from most likely to least likely to be flaky.
+        for (int i = 0;
+            i < GenInputsAbstract.nondeterministic_methods_to_output
```

Comment:
This would be clearer if you put the max in a variable.  Or you just set `nondeterministic_methods_to_output`, if it is guaranteed not to be used or output again.